src/trace.jl

```diff
@@ -557,10 +557,38 @@ function trace_loops!(ir::IR)
 end
 
 
+
+function expand!(ir::IR)
+    # TODO delete this function
+    # Variables orders may be switched in IRTools.expand!
+    # depending on details of the julia version/environment.
+    # This bandaid forces a uniform order
+    # See also https://github.com/dfdx/Ghost.jl/pull/7
+    worklist = IRTools.blocks(ir)
+    spats = Dict(b => Dict() for b in IRTools.blocks(ir))
+    while !isempty(worklist)
+      b = pop!(worklist)
+      b.id == 1 && continue
+      defs = IRTools.definitions(b)
+      uses = IRTools.usages(b)
+      uses = sort(collect(uses), by=v->v.id, rev=true)
```

Comment:
I could have sworn this worked without the `rev=true`, but today it needs `rev=true`.

Reply:
This frightens me a bit. If the behavior isn't consistent across sessions, we may end up with silent non-reproducible bugs. I think at the moment this PR is the best option, but after Julia 1.7 is officially released I'm going to revisit the issue and make additional checks.

Reply:
Yes, I am also a bit worried. But I think the most efficient way forward is to try it out and see what happens.